homeassistant/components/device_tracker/owntracks.py

```diff
@@ -56,15 +63,63 @@ def setup_scanner(hass, config, see):
     max_gps_accuracy = config.get(CONF_MAX_GPS_ACCURACY)
     waypoint_import = config.get(CONF_WAYPOINT_IMPORT)
     waypoint_whitelist = config.get(CONF_WAYPOINT_WHITELIST)
+    secret = config.get(CONF_SECRET)
 
-    def validate_payload(payload, data_type):
+    def decrypt_payload(topic, ciphertext):
+        """Decrypt encrypted payload."""
+        try:
```

Comment:
added get_cipher() in homeassistant/components/device_tracker/owntracks.py which is mocked in tests/components/device_tracker/test_owntracks.py